lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/CompositeProxy.java

```diff
@@ -167,15 +168,15 @@ public C moveToBottom() {
     @Override
     public Map<ControlHandleType, IControlHandleList> getControlHandles(List<ControlHandleType> types) {
         if ((null == types) || (types.isEmpty())) {
-            return null;
+            return new HashMap<>();
         }
         if (types.size() > 1) {
             types = new ArrayList<ControlHandleType>(new HashSet<ControlHandleType>(types));
         }
         IControlHandleFactory factory = getControlHandleFactory();
 
         if (null == factory) {
-            return null;
+            return new HashMap<>();
```

Comment:
Hi @jomarko,

did you check all `getControlHandles` method invocations? If I correct some of them already relies on `null`s so this small change can lead to further issues.

Reply:
same here